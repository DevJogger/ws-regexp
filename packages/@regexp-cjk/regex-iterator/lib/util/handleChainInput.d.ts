import { IChainInput, IChainInputObject, IChainInputArray } from '../types';
import { ICloneRegexpOptions } from '@regexp-cjk/clone-regexp';
export declare function handleChainInputCore2(row: IChainInput | IChainInputArray, options?: ICloneRegexpOptions): [RegExp, string | ((substring: string, ...args: string[]) => string)];
export declare function handleChainInputCore(row: IChainInput, options?: ICloneRegexpOptions): Required<IChainInputObject>;
export declare function handleChainInput(row: IChainInput, options: ICloneRegexpOptions): Required<IChainInputObject>;
export default handleChainInput;
