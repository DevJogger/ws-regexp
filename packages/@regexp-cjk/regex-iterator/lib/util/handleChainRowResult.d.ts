export declare function handleChainRowResult({ regexp, backref, match, allowFallbackToSource, }: {
    regexp: RegExp;
    backref: number | string;
    match: RegExpMatchArray;
    allowFallbackToSource: boolean;
}, str: string): string;
export default handleChainRowResult;
