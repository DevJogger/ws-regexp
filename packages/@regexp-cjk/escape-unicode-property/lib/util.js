"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = void 0;
const util_1 = require("@regexp-cjk/rewrite-pattern/lib/util");
const lodash_1 = require("lodash");
function handleOptions(options, flags) {
    return (0, util_1.handleOptions)((0, lodash_1.merge)({}, options, {
        unicodePropertyEscape: true,
        rewriteFlags: {
            unicode: true,
        },
    }), flags !== null && flags !== void 0 ? flags : 'u');
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map