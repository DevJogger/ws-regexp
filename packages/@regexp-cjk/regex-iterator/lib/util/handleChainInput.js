"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleChainInput = exports.handleChainInputCore = exports.handleChainInputCore2 = void 0;
const tslib_1 = require("tslib");
const clone_regexp_1 = (0, tslib_1.__importDefault)(require("@regexp-cjk/clone-regexp"));
function handleChainInputCore2(row, options) {
    let regexp;
    let replaceValue;
    if (row instanceof RegExp) {
        regexp = row;
    }
    else if (Array.isArray(row)) {
        ([
            regexp,
            replaceValue,
        ] = row);
    }
    else {
        ({
            regexp,
            replaceValue,
        } = row);
    }
    return [regexp, replaceValue];
}
exports.handleChainInputCore2 = handleChainInputCore2;
function handleChainInputCore(row, options) {
    let regexp;
    let backref = 0;
    let replaceValue;
    if (row instanceof RegExp) {
        regexp = row;
    }
    else {
        ({
            regexp,
            backref,
            replaceValue,
        } = row);
    }
    return {
        regexp,
        backref,
        replaceValue,
    };
}
exports.handleChainInputCore = handleChainInputCore;
function handleChainInput(row, options) {
    let ret = handleChainInputCore(row, options);
    ret.regexp = (0, clone_regexp_1.default)(ret.regexp, options);
    return ret;
}
exports.handleChainInput = handleChainInput;
exports.default = handleChainInput;
//# sourceMappingURL=handleChainInput.js.map