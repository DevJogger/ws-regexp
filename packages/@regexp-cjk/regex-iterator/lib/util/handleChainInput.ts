import { IChainInput, IChainInputObject, IChainInputArray } from '../types';
import cloneRegexp, { ICloneRegexpOptions } from '@regexp-cjk/clone-regexp';

export function handleChainInputCore2(row: IChainInput | IChainInputArray, options?: ICloneRegexpOptions): [RegExp, string | ((substring: string, ...args: string[]) => string)]
{
	let regexp: RegExp;
	let replaceValue: string | ((substring: string, ...args: string[]) => string);

	if (row instanceof RegExp)
	{
		regexp = row;
	}
	else if (Array.isArray(row))
	{
		([
			regexp,
			replaceValue,
		] = row);
	}
	else
	{
		({
			regexp,
			replaceValue,
		} = row);
	}

	return [regexp, replaceValue]
}


export function handleChainInputCore(row: IChainInput, options?: ICloneRegexpOptions): Required<IChainInputObject>
{
	let regexp: RegExp;
	let backref: number | string = 0;
	let replaceValue: string | ((substring: string, ...args: string[]) => string);

	if (row instanceof RegExp)
	{
		regexp = row;
	}
	else
	{
		({
			regexp,
			backref,
			replaceValue,
		} = row);
	}

	return {
		regexp,
		backref,
		replaceValue,
	}
}

export function handleChainInput(row: IChainInput, options: ICloneRegexpOptions)
{
	let ret = handleChainInputCore(row, options)

	ret.regexp = cloneRegexp(ret.regexp, options);

	return ret
}

export default handleChainInput
