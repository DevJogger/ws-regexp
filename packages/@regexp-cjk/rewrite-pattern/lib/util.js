"use strict";
/**
 * Created by user on 2020/6/8.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = exports.rewritePatternOptions = void 0;
const lodash_1 = require("lodash");
const rewrite_flags_1 = require("@regexp-cjk/rewrite-flags");
function rewritePatternOptions(options, flags) {
    var _a;
    return (0, lodash_1.defaults)({}, options, {
        dotAllFlag: true,
        lookbehind: true,
        namedGroup: true,
        useUnicodeFlag: (_a = flags === null || flags === void 0 ? void 0 : flags.includes) === null || _a === void 0 ? void 0 : _a.call(flags, 'u'),
    });
}
exports.rewritePatternOptions = rewritePatternOptions;
function handleOptions(options, flags) {
    var _a;
    flags = (_a = options === null || options === void 0 ? void 0 : options.flags) !== null && _a !== void 0 ? _a : flags;
    if (options === null || options === void 0 ? void 0 : options.rewriteFlags) {
        flags = (0, rewrite_flags_1.rewriteFlags)(flags, options.rewriteFlags);
    }
    options = rewritePatternOptions(options, flags);
    delete options.flags;
    return {
        options,
        flags,
    };
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map