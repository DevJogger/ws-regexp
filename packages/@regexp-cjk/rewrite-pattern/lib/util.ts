/**
 * Created by user on 2020/6/8.
 */

import { defaults } from 'lodash';
import { IOptionsRewritePattern } from '../index';
import { rewriteFlags } from '../flags';

export function rewritePatternOptions(options?: IOptionsRewritePattern, flags?: string): IOptionsRewritePattern
{
	return defaults<IOptionsRewritePattern, IOptionsRewritePattern, IOptionsRewritePattern>({}, options, {
		dotAllFlag: true,
		lookbehind: true,
		namedGroup: true,
		useUnicodeFlag: flags?.includes?.('u'),
	})
}

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	flags = options?.flags ?? flags;

	if (options?.rewriteFlags)
	{
		flags = rewriteFlags(flags, options.rewriteFlags);
	}

	options = rewritePatternOptions(options, flags)

	return {
		options,
		flags,
	}
}
