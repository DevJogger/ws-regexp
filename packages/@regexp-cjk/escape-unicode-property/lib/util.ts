import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
import { handleOptions as _handleOptions } from '@regexp-cjk/rewrite-pattern/lib/util';
import { merge } from 'lodash';

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	return _handleOptions(merge({}, options, {
		unicodePropertyEscape: true,
		rewriteFlags: {
			unicode: true,
		},
	}), flags ?? 'u');
}
