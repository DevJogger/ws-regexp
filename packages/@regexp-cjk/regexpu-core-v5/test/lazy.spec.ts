// @ts-ignore
import _rewritePattern from 'regexpu-core';
import rewritePatternV5 from '../src/index';
import { inspect } from 'util';

(<Parameters<typeof rewritePatternV5>[]>[

	[
		'\\u{ab}', '', {
		unicodeFlag: 'transform',
	},
	],

	[
		'\\u{ab}', 'u', {
		unicodeFlag: 'transform',
	},
	],

	[
		'.', '', {
		dotAllFlag: 'transform',
	},
	],

	[
		'.', 's', {
		dotAllFlag: 'transform',
	},
	],

	[
		'.', 'su', {
		dotAllFlag: 'transform',
	},
	],

	[
		'\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
		unicodePropertyEscapes: 'transform',
	},
	],

	[
		'\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
		unicodeFlag: 'transform',
		unicodePropertyEscapes: 'transform',
	},
	],

	[
		'(?<name>.)\\k<name>', '', {
		namedGroups: "transform",
	},
	],

	[
		'First_Name: (?<firstname>\\w+), Last_Name: (?<lastname>\\w+)', '', {

	},
	],

	[
		'First_Name: (?<firstname>\\w+), Last_Name: (?<lastname>\\w+)', '', {
namedGroups: "transform",
	},
	],

	[
		'First_Name: (?<firstname>\\w+), Last_Name: (?<lastname>\\w+)\\k<firstname>', '', {
namedGroups: "transform",
	},
	],

	[
		'[\\p{Emoji}&&\\p{ASCII}]', 'u', {
		unicodeSetsFlag: 'transform',
	},
	],

	[
		'[^[a-h]&&[f-z]]', 'v', {
		unicodeSetsFlag: 'transform',
	},
	],

	[
		'[^[a-h]&&[f-z]]', 'v', {
		unicodeSetsFlag: 'transform',
		unicodeFlag: 'transform',
	},
	],

]).forEach(argv =>
{
	test(inspect(argv), () =>
	{

		let actual = rewritePatternV5(...argv);
		let expected = _rewritePattern(...argv);

		console.dir(actual);

		expect(actual).toStrictEqual(expected);

		expect({
			actual,
			changed: actual !== argv[0],
		}).toMatchSnapshot();

	});
});
