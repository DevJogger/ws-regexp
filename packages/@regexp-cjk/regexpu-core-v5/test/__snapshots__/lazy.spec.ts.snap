// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[
  '[^[a-h]&&[f-z]]',
  'v',
  { unicodeSetsFlag: 'transform', unicodeFlag: 'transform' }
] 1`] = `
{
  "actual": "(?:(?![f-h])[\\s\\S])",
  "changed": true,
}
`;

exports[`[
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodeFlag: 'transform', unicodePropertyEscapes: 'transform' }
] 1`] = `
{
  "actual": "(?:\\uD811[\\uDC00-\\uDE46])",
  "changed": true,
}
`;

exports[`[
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscapes: 'transform' }
] 1`] = `
{
  "actual": "[\\u{14400}-\\u{14646}]",
  "changed": true,
}
`;

exports[`[
  'First_Name: (?<firstname>\\\\w+), Last_Name: (?<lastname>\\\\w+)',
  '',
  { namedGroups: 'transform' }
] 1`] = `
{
  "actual": "First_Name: (\\w+), Last_Name: (\\w+)",
  "changed": true,
}
`;

exports[`[
  'First_Name: (?<firstname>\\\\w+), Last_Name: (?<lastname>\\\\w+)',
  '',
  {}
] 1`] = `
{
  "actual": "First_Name: (?<firstname>\\w+), Last_Name: (?<lastname>\\w+)",
  "changed": false,
}
`;

exports[`[
  'First_Name: (?<firstname>\\\\w+), Last_Name: (?<lastname>\\\\w+)\\\\k<firstname>',
  '',
  { namedGroups: 'transform' }
] 1`] = `
{
  "actual": "First_Name: (\\w+), Last_Name: (\\w+)\\1",
  "changed": true,
}
`;

exports[`[ '(?<name>.)\\\\k<name>', '', { namedGroups: 'transform' } ] 1`] = `
{
  "actual": "(.)\\1",
  "changed": true,
}
`;

exports[`[ '.', '', { dotAllFlag: 'transform' } ] 1`] = `
{
  "actual": ".",
  "changed": false,
}
`;

exports[`[ '.', 's', { dotAllFlag: 'transform' } ] 1`] = `
{
  "actual": "[\\s\\S]",
  "changed": true,
}
`;

exports[`[ '.', 'su', { dotAllFlag: 'transform' } ] 1`] = `
{
  "actual": "[\\s\\S]",
  "changed": true,
}
`;

exports[`[ '[\\\\p{Emoji}&&\\\\p{ASCII}]', 'u', { unicodeSetsFlag: 'transform' } ] 1`] = `
{
  "actual": "[\\p{Emoji}&&\\p{ASCII}]",
  "changed": false,
}
`;

exports[`[ '[^[a-h]&&[f-z]]', 'v', { unicodeSetsFlag: 'transform' } ] 1`] = `
{
  "actual": "[^f-h]",
  "changed": true,
}
`;

exports[`[ '\\\\u{ab}', '', { unicodeFlag: 'transform' } ] 1`] = `
{
  "actual": "u\\{ab\\}",
  "changed": true,
}
`;

exports[`[ '\\\\u{ab}', 'u', { unicodeFlag: 'transform' } ] 1`] = `
{
  "actual": "\\xAB",
  "changed": true,
}
`;
